falcon.go

```diff
@@ -36,96 +45,131 @@ const (
 	// CurrentSaltVersion is the salt version number used to compute signatures.
 	// The salt version is incremented when the signing procedure changes (rarely).
 	CurrentSaltVersion = C.FALCON_DET1024_CURRENT_SALT_VERSION
+	// CTSignatureSize is the max size in bytes of a Falcon signature in CT format
+	CTSignatureSize = C.FALCON_DET1024_SIG_CT_SIZE
+	// SignatureMaxSize is the max possible size in bytes of a Falcon signature in a compressed format.
+	SignatureMaxSize = C.FALCON_DET1024_SIG_COMPRESSED_MAXSIZE
 )
 
-type PublicKey []byte
-type PrivateKey []byte
+type PublicKey [PublicKeySize]byte
+type PrivateKey [PrivateKeySize]byte
 
 // CompressedSignature is a deterministic Falcon signature in compressed
 // form, which is variable-length.
 type CompressedSignature []byte
 
 // CTSignature is a deterministic Falcon signature in constant-time form,
 // which is fixed-length.
-type CTSignature []byte
+type CTSignature [CTSignatureSize]byte
 
 // GenerateKey generates a public/private key pair from the given seed.
 func GenerateKey(seed []byte) (PublicKey, PrivateKey, error) {
 	var rng C.shake256_context
-	C.shake256_init_prng_from_seed(&rng, unsafe.Pointer(&seed[0]), C.size_t(len(seed)))
 
-	publicKey := make([]byte, PublicKeySize)
-	privateKey := make([]byte, PrivateKeySize)
+	seedLen := len(seed)
+	seedData := (*C.uchar)(C.NULL)
```

Comment:
It took me a moment to realize C.NULL is supplied not by cgo but by the NULL macro in C. does the zero value of e.g. `var seedData *C.uchar` not provide the same value?

Reply:
it might be right. I don't know.
I think it is more clear that we pass NULL to the function.
any other idea?

Reply:
Not really. I imagine either the zero value or C.NULL should end up the same.